File: Test/app/src/main/java/com/android/hootr/test/view/login/LoginActivity.java
package com.android.hootr.test.view.login;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.android.hootr.test.App;
import com.android.hootr.test.R;
import com.android.hootr.test.data.api.Api1C;
import com.android.hootr.test.data.prefs.Prefs;
import com.android.hootr.test.view.LaunchActivity;

public class LoginActivity extends AppCompatActivity implements LoginView {

    private Button buttonLogin;
    private EditText editLogin;
    private ViewGroup progress;

    private LoginPresenter presenter;

    public static void start(Context context) {
        Intent intent = new Intent(context, LoginActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);


        Prefs prefs = ((App) getApplication()).getPrefs();
        Api1C api1C = ((App) getApplication()).getApi1C();

        presenter = new LoginPresenterImpl(prefs, api1C);

        editLogin = findViewById(R.id.edit_cod_user);
        buttonLogin = findViewById(R.id.buttonLogin);
        progress = findViewById(R.id.progress);

        buttonLogin.setOnClickListener(v -> {
            presenter.loginUser(editLogin.getText().toString().trim());
        });

        presenter.attachView(this);

    }

    @Override
    protected void onDestroy() {
        presenter.detachView();
        super.onDestroy();
    }

    @Override
    public void showError(String error) {
        Toast.makeText(this, error, Toast.LENGTH_LONG).show();
    }

    @Override
    public void navigateToLaunchScreen() {
        LaunchActivity.start(this);
    }

    @Override
    public void showProgress() {
        progress.setVisibility(View.VISIBLE);
    }

    @Override
    public void hideProgress() {
        progress.setVisibility(View.GONE);
    }
}

File: TSD/app/src/main/java/com/android/hootr/tsd/screens/login/LoginActivity.kt
package com.android.hootr.tsd.screens.login

import android.content.Context
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.view.KeyEvent
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import com.android.hootr.tsd.App
import com.android.hootr.tsd.R
import com.android.hootr.tsd.api.login.LoginApi
import com.android.hootr.tsd.screens.LaunchActivity
import com.android.hootr.tsd.screens.event.login.EventLogin
import com.android.hootr.tsd.useCase.showKeyboardEditTextEx
import kotlinx.android.synthetic.main.activity_login.*

class LoginActivity : AppCompatActivity() {

    var viewModel: LoginViewModel? = null
    lateinit var loginApi: LoginApi

    companion object {
        fun startActivity(context: Context) {
            val starter = Intent(context, LoginActivity::class.java)
//        starter.flags = (Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
            context.startActivity(starter)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        (application as App).applicationComponent.inject(this)

        initViewModel()

        editTextCodeUser.requestFocus()

        showKeyboardEditTextEx(editTextCodeUser)

        editTextCodeUser.setOnKeyListener { view, i, keyEvent ->

            if (keyEvent.action == KeyEvent.ACTION_DOWN
                && i == KeyEvent.KEYCODE_ENTER
            ) {
                btnLogin.requestFocus()
                editTextCodeUser.isCursorVisible = false
                true
            } else false

        }

        viewModel?.event?.observe(this, androidx.lifecycle.Observer {
            when (it) {
                is EventLogin.Success -> success()
                is EventLogin.Loading -> showLoading()
                is EventLogin.ShowError -> showError(it.messeg)
            }

        })
    }

    fun initViewModel() {
        viewModel = ViewModelProvider(this).get(LoginViewModel::class.java)
    }

    private fun showError(messeg: String) {
        progress.visibility = View.INVISIBLE
        btnLogin.isEnabled = false
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            editTextCodeUser.focusable = View.FOCUSABLE
        }
        editTextCodeUser.isLongClickable = true
        editTextCodeUser.isCursorVisible = true

        Toast.makeText(this, messeg, Toast.LENGTH_LONG).show()
    }

    fun success() {
        progress.visibility = View.INVISIBLE
        LaunchActivity.startActivity(this)
    }

    fun showLoading() {
        progress.visibility = View.GONE
        btnLogin.isEnabled = false
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            editTextCodeUser.focusable = View.NOT_FOCUSABLE
        }
        editTextCodeUser.isLongClickable = false
        editTextCodeUser.isCursorVisible = false

    }


}
